locket/__init__.py

```diff
@@ -10,22 +10,35 @@
     import fcntl
 except ImportError:
     try:
+        import ctypes
+        import ctypes.wintypes
         import msvcrt
     except ImportError:
-        raise ImportError("Platform not supported (failed to import fcntl, msvcrt)")
+        raise ImportError("Platform not supported (failed to import fcntl, ctypes, msvcrt)")
     else:
+        _WinAPI_LockFile = ctypes.windll.kernel32.LockFile
+        _WinAPI_LockFile.restype = ctypes.wintypes.BOOL
+        _WinAPI_LockFile.argtypes = [ctypes.wintypes.HANDLE] + [ctypes.wintypes.DWORD] * 4
+
+        _WinAPI_UnlockFile = ctypes.windll.kernel32.UnlockFile
+        _WinAPI_UnlockFile.restype = ctypes.wintypes.BOOL
+        _WinAPI_UnlockFile.argtypes = [ctypes.wintypes.HANDLE] + [ctypes.wintypes.DWORD] * 4
+
         _lock_file_blocking_available = False
 
         def _lock_file_non_blocking(file_):
-            try:
-                msvcrt.locking(file_.fileno(), msvcrt.LK_NBLCK, 1)
+            res = _WinAPI_LockFile(msvcrt.get_osfhandle(file_.fileno()), 0, 0, 1, 0)
+            if res:
                 return True
-            # TODO: check errno
-            except IOError:
+            else:
+                e = ctypes.WinError()
```

Comment:
With `use_last_error` enabled, this should be `err = ctypes.get_last_error()`. Then you can raise `ctypes.WinError(err)`. Creating the exception without an explicit error value will implicitly call `GetLastError` via FFI, which is not always reliable in a high-level environment, such as when working directly with a ctypes wrapped function in the REPL.

Reply:
@eryksun Thanks for the insight. It would be nice to document those best practices in the ctypes official docs, if you feel like it.